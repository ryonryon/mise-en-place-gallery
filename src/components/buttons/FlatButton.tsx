import styled from "@emotion/styled";
import * as React from "react";
import { secondary, white } from "../../constants/colors";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  width: string;
}

const FlatButton = ({ width, ...rest }: Props) => {
  const FlatStyledButton = styled.button`
    background-color: ${white};
    border: solid 1px ${secondary};
    border-radius: 5px;
    color: ${secondary};
    height: 30px;
    width: ${width};
    cursor: pointer;
    margin: 16px;
    padding: 4px;
  `;

  return <FlatStyledButton {...rest}></FlatStyledButton>;
};

FlatButton.defaultProps = {
  width: "150px"
};

export default FlatButton;
