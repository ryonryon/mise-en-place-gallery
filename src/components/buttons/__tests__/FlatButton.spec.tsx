import * as React from "react";
import * as renderer from "react-test-renderer";

import FlatButton from "../FlatButton";

describe("FlatButton test", () => {
  it("tests FlatButton renders correctly ", () => {
    const flatButton = renderer.create(<FlatButton />).toJSON();
    expect(flatButton).toMatchSnapshot();
  });
});
