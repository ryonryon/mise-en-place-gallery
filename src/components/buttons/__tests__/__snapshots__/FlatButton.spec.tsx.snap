// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatButton test tests FlatButton renders correctly  1`] = `
<button
  className="css-15gv13u"
/>
`;
